File: code/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/template/formfield/LabelWidthInPixelLookupCall.java
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.scout.widgets.client.ui.template.formfield;

import java.util.ArrayList;
import java.util.List;

import org.eclipse.scout.rt.client.ui.form.fields.IFormField;
import org.eclipse.scout.rt.platform.classid.ClassId;
import org.eclipse.scout.rt.shared.services.lookup.ILookupRow;
import org.eclipse.scout.rt.shared.services.lookup.LocalLookupCall;
import org.eclipse.scout.rt.shared.services.lookup.LookupRow;

@ClassId("d6f889d7-eb32-4061-b831-77bc7e8423f7")
public class LabelWidthInPixelLookupCall extends LocalLookupCall<String> {

  private static final long serialVersionUID = 1L;

  @Override
  protected List<? extends ILookupRow<String>> execCreateLookupRows() {
    List<ILookupRow<String>> rows = new ArrayList<>();
    rows.add(new LookupRow<>(IFormField.LABEL_WIDTH_DEFAULT + "", "default"));
    rows.add(new LookupRow<>(IFormField.LABEL_WIDTH_UI + "", "ui width"));
    return rows;
  }
}

File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/formfield/LabelWidthInPixelLookupCall.ts
/*
 * Copyright (c) 2010, 2023 BSI Business Systems Integration AG
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 */
import {FormField, FormFieldLabelWidth, StaticLookupCall} from '@eclipse-scout/core';

export class LabelWidthInPixelLookupCall extends StaticLookupCall<FormFieldLabelWidth> {

  constructor() {
    super();
  }

  protected override _data(): any[] {
    return LabelWidthInPixelLookupCall.DATA;
  }

  static DATA = [
    [FormField.LabelWidth.DEFAULT, 'default'],
    [FormField.LabelWidth.UI, 'ui width']
  ];
}
